pkg/api/validation/validation.go

```diff
@@ -1241,6 +1263,13 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 	}
 
 	switch secret.Type {
+	case api.SecretTypeServiceAccountToken:
```

Comment:
right now, the token controller will:
1. create new ServiceAccountToken secrets with the token data filled in
2. add token data to ServiceAccountToken secrets that are missing it, to enable creation of new tokens

requiring token data in validation would prevent number 2